app/src/main/java/org/p2p/wallet/auth/interactor/MetadataInteractor.kt

```diff
@@ -17,9 +17,13 @@ class MetadataInteractor(
     private val tokenKeyProvider: TokenKeyProvider,
     private val seedPhraseProvider: SeedPhraseProvider,
     private val secureStorageContract: SecureStorageContract,
+    private val gatewayMetadataMerger: GatewayMetadataMerger,
 ) {
 
-    var currentMetadata: GatewayOnboardingMetadata? = null
+    var currentMetadata: GatewayOnboardingMetadata? = secureStorageContract.getObject(
```

Comment:
better to do get() = here
it can fail on interactor creation and never get the new value